Firestore/core/test/unit/remote/serializer_test.cc

```diff
@@ -1742,6 +1742,60 @@ TEST_F(SerializerTest, DecodesVersionWithTargets) {
   ExpectDeserializationRoundTrip(model, proto);
 }
 
+TEST_F(SerializerTest, GracefullyDecodesBadMapValue) {
+  pb_bytes_array_t key{1, {'a'}};
+  google_firestore_v1_MapValue_FieldsEntry fields_entry;
```

Comment:
Nit: it seems a little inconsistent that some of these C structs are initialized to zeros and some aren't. Note also that for simple structs with few members, you should be able to use aggregate initialization:
```cpp
google_firestore_v1_Value bad_value = {};
google_firestore_v1_MapValue_FieldsEntry fields_entry = {&key, bad_value};
```

Feel free to ignore since this test will likely go away.

Reply:
I have run into a fair number of problems when I didn't initialize the value to zero. I think we should fix this.

Reply:
Done